ui/components/app/flask/update-snap-permission-list/update-snap-permission-list.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { getPermissionDescription } from '../../../../helpers/utils/permission';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+
+export default function UpdateSnapPermissionList({
+  approvedPermissions,
+  revokedPermissions,
+  newPermissions,
+}) {
+  const t = useI18nContext();
+
+  const ApprovedPermissions = () => {
+    return Object.keys(approvedPermissions).map((approvedPermission) => {
+      const { label, rightIcon } = getPermissionDescription(
+        t,
+        approvedPermission,
+      );
+      const { date } = approvedPermissions[approvedPermission];
+      const dateObj = new Date(date);
+      const [month, day, year] = [
+        dateObj.getMonth(),
+        dateObj.getDate(),
+        dateObj.getFullYear(),
+      ];
+      // getMonth is zero-based
+      const formattedDate = `${year}-${month + 1}-${day}`;
```

Comment:
I think we should use `luxon` for date formatting. We do this elsewhere as well.